File: apps/windows/TeslaSync.App.Core/Push/PushCapabilities.cs
using System.Collections.ObjectModel;

namespace TeslaSync.App.Core.Push;

/// <summary>
/// Identifiers and notification-capability flags reported to the backend at device-registration
/// time (P2/W6-0002). The platform/provider strings name the push transport; the capability flags
/// describe what kinds of notification this client can present so the <c>notification-worker</c> can
/// tailor fan-out (ADR-009). All values are static, non-PII descriptors.
/// </summary>
public static class PushCapabilities
{
    /// <summary>The platform identifier sent as <c>platform</c> in the registration payload.</summary>
    public const string WindowsPlatform = "windows";

    /// <summary>The push transport identifier sent as <c>push_provider</c> (Windows Notification Service).</summary>
    public const string WnsProvider = "wns";

    /// <summary>Capability: the client can present interruptive toast notifications.</summary>
    public const string Toast = "toast";

    /// <summary>Capability: the client can render badge/tile counters.</summary>
    public const string Badge = "badge";

    /// <summary>Capability: the client can surface in-app alert banners while in the foreground.</summary>
    public const string Alert = "alert";

    /// <summary>The default Windows capability set (toast + badge + alert).</summary>
    public static readonly IReadOnlyList<string> WindowsDefault =
        new ReadOnlyCollection<string>(new[] { Toast, Badge, Alert });
}

File: apps/android/app/src/main/java/io/teslasync/android/push/PushCapabilities.kt
package io.teslasync.android.push

/**
 * Identifiers and notification-capability flags reported to the backend at device-registration time
 * (P3/A6, ADR-009). The platform/provider strings name the push transport; the capability flags
 * describe what kinds of notification this client can present so the `notification-worker` can tailor
 * fan-out. All values are static, non-PII descriptors.
 */
object PushCapabilities {
    /** The platform identifier sent as `platform` in the registration payload. */
    const val ANDROID_PLATFORM = "android"

    /** The push transport identifier sent as `push_provider` (Firebase Cloud Messaging). */
    const val FCM_PROVIDER = "fcm"

    /** Capability: the client can present system notifications in the tray / lock screen. */
    const val NOTIFICATION = "notification"

    /** Capability: the client can render badge counters. */
    const val BADGE = "badge"

    /** Capability: the client can surface in-app alert banners while in the foreground. */
    const val ALERT = "alert"

    /** The default Android capability set (system notification + badge + in-app alert). */
    val ANDROID_DEFAULT: List<String> = listOf(NOTIFICATION, BADGE, ALERT)
}
